pkg/controller/jobframework/interface.go

```diff
@@ -119,6 +119,13 @@ type JobWithCustomValidation interface {
 	ValidateOnUpdate(oldJob GenericJob) (field.ErrorList, error)
 }
 
+// JobWithCustomWorkloadRetriever interface should be implemented by jobs which have
+// their own logic to get workload, e.g. for RayJob with InTreeAutoScaling, the workload
+// is attached on the underlying RayCluster, RayJob needs to get workload through that RayCluster
+type JobWithCustomWorkloadRetriever interface {
```

Comment:
I think this interface is no longer needed if we make the submitter Job top-level as mentioned here: https://github.com/kubernetes-sigs/kueue/pull/7769/files#r2588885609